public/apps/configuration/panels/user-list.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageHeader,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { difference, map, Dictionary } from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { AppDependencies } from '../../types';
+import {
+  getUserList,
+  InternalUsersListing,
+  requestDeleteUsers,
+} from '../utils/internal-user-list-utils';
+
+function dictView() {
+  return (items: Dictionary<string>) => {
+    return (
+      <EuiFlexGroup direction="column" style={{ margin: '1px' }}>
+        {map(items, (v, k) => (
+          <EuiText key={k} size="xs">
+            {k}: {v}
+          </EuiText>
+        ))}
+      </EuiFlexGroup>
+    );
+  };
+}
+
+const columns = [
+  {
+    field: 'username',
+    name: 'Username',
+    sortable: true,
+  },
+  {
+    field: 'attributes',
+    name: 'Attributes',
+    render: dictView(),
+    truncateText: true,
+  },
+];
+
+export function UserList(props: AppDependencies) {
+  const [userData, setUserData] = useState<InternalUsersListing[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<InternalUsersListing[]>([]);
+  const [isActionsPopoverOpen, setActionsPopoverOpen] = useState(false);
+  const [currentUsername, setCurrentUsername] = useState('');
+  // TODO: fetch current username
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setUserData(await getUserList(props.coreStart.http));
+      } catch (e) {
+        console.log(e);
+        setErrorFlag(true);
+      }
+    };
+
+    fetchData();
+  }, [props.coreStart.http]);
+
+  const handleDelete = async () => {
+    const usersToDelete: string[] = selection.map((r) => r.username);
+    try {
+      await requestDeleteUsers(props.coreStart.http, usersToDelete);
+      // Refresh from server (calling fetchData) does not work here, the server still return the roles
+      // that had been just deleted, probably because ES takes some time to sync to all nodes.
+      // So here remove the selected roles from local memory directly.
+      setUserData(difference(userData, selection));
```

Comment:
Just want to confirm that if user quickly refreshes the page (before ES deletion is done), that deleted user will show up again.

Reply:
In the edge case, yes.